dvc/remote/gs.py

```diff
@@ -22,33 +25,49 @@ def wrapper(*args, **kwargs):
         import requests
         from google.cloud.storage.blob import Blob, _DEFAULT_CHUNKSIZE
 
-        # Default chunk size for gs is 100M, which might be too much for
-        # particular network (see [1]). So if we are getting ConnectionError,
-        # we should try lowering the chunk size until we reach the minimum
-        # allowed chunk size of 256K. Also note that `chunk_size` must be a
-        # multiple of 256K per the API specification.
+        # `ConnectionError` may be due to too large `chunk_size`
+        # (see [#2572]) so try halving on error.
+        # Note: default 100M is too large for fine-grained progress
+        # so 10M is the starting default.
+        # Note: minimum 256K.
+        # Note: must be multiple of 256K.
         #
-        # [1] https://github.com/iterative/dvc/issues/2572
+        # [#2572]: https://github.com/iterative/dvc/issues/2572
 
         # skipcq: PYL-W0212
-        multiplier = int(_DEFAULT_CHUNKSIZE / Blob._CHUNK_SIZE_MULTIPLE)
+        multiplier = int(
+            _DEFAULT_CHUNKSIZE / (10.0 * Blob._CHUNK_SIZE_MULTIPLE)
+        )
         while True:
             try:
                 # skipcq: PYL-W0212
                 chunk_size = Blob._CHUNK_SIZE_MULTIPLE * multiplier
                 return func(*args, chunk_size=chunk_size, **kwargs)
             except requests.exceptions.ConnectionError:
-                multiplier = int(multiplier / 2)
+                multiplier = multiplier // 2
                 if not multiplier:
                     raise
 
     return wrapper
 
 
 @dynamic_chunk_size
-def _upload_to_bucket(bucket, from_file, to_info, **kwargs):
-    blob = bucket.blob(to_info.path, **kwargs)
-    blob.upload_from_filename(from_file)
+def _upload_to_bucket(bucket, from_file, to_info, chunk_size=None, **kwargs):
+    blob = bucket.blob(to_info.path, chunk_size=chunk_size, **kwargs)
+    with Tqdm(
+        desc=to_info.path, total=os.path.getsize(from_file), bytes=True
```

Comment:
seemed best to use `desc=to_info.path` but maybe there are better options?

Reply:
@casperdcl Sure, take a look at s3.py, we are using `name` there.

Reply:
Also you've forgotten the `disable` flag.

Reply:
yes I was wondering if `kwargs` contained something along the lines of `no_progress`. Also saw the `name` thing. Will fix in a few hours.